### PR TITLE
Feature/search by ingredients

### PR DESCRIPTION
**Pull request by apasashkov**
Search by ingredients feature.

### CODE REVIEW

File: controllers/recipeController.js [hunk 1]
```diff
@@ -422,4 +422,35 @@ recipeController.updateRecipe = (req, res) => {
...
+        } else {
+            res.json(result.rows);
+        }
+    });
+};
+
+recipeController.getRecepiesByIngredients = (req, res) => {
+    const ingredients = req.params.ingredients;
+    if (!ingredients) {
+        res.sendStatus(500);
```
Comment (python4eg) [inline controllers/recipeController.js]: not 500 but 404. 500 is a server side error

File: static/js/pages/RecipesPage/SearchComponent/MultiSelect/MultiSelect.js [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+            isOpen: false,
+            selectedValues: [],
+            filteredOptions: [],
+        };
+    }
+
+    componentDidUpdate() {
+        const diff = this.wrapperContainer.offsetWidth - this.wrapper.offsetWidth;
+        if (diff > 0) {
+            this.wrapperScrollable.style.left = `-${diff}px`;
```
Comment (python4eg) [inline static/js/pages/RecipesPage/SearchComponent/MultiSelect/MultiSelect.js]: nit pick: You can use ternary operator here

File: static/js/pages/RecipesPage/SearchComponent/MultiSelect/Option.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class Option extends Component {
+    static propTypes = {
```
Comment (python4eg) [inline static/js/pages/RecipesPage/SearchComponent/MultiSelect/Option.js]: Should be defined outside class
e.g.
```
class Option extends Component {
...
}
Option.propTypes = {
...
}
```

File: static/js/pages/RecipesPage/SearchComponent/MultiSelect/Option.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        value: PropTypes.string.isRequired,
+        onSelect: PropTypes.func.isRequired,
+    }
+
+    static defaultProps = {
```
Comment (python4eg) [inline static/js/pages/RecipesPage/SearchComponent/MultiSelect/Option.js]: same here

File: static/js/pages/RecipesPage/SearchComponent/SearchComponent.js [hunk 5]
```diff
@@ -26,25 +27,51 @@ class SearchComponent extends Component {
...
     componentDidUpdate(prevProps, prevState) {
         if (prevState.item !== this.state.item && this.state.item) {
             this.setState({ process: 'fetching' });
         }
-        
+
+        if (prevState.type !== this.state.type && this.state.type === 'searchByIngredients') {
+            axios.get('/api/recipes/getAllIngredients')
+            .then((response) => {
+                const sortedData = response.data.sort((a, b) => {
```
Comment (python4eg) [inline static/js/pages/RecipesPage/SearchComponent/SearchComponent.js:50]: :+1: and You can use underscore's `_.sortBy`

**Review by StasMelnyk — APPROVED** [review]

**Pull request merged**